ClipboardZanager/Sources/ClipboardZanager.Core.Desktop/Services/WindowsService.cs

```diff
@@ -256,6 +256,34 @@ private Window GetWindowInformation(IntPtr windowHandle)
                 icon = GetWin32WindowIcon(windowHandle, applicationIdentifier);
             }
 
+            if (icon == null)
+            {
+                icon = new BitmapImage();
+                if (applicationIdentifier == Environment.GetFolderPath(Environment.SpecialFolder.Windows) + "\\explorer.exe" && stringBuilder.ToString() == "") // Desktop
+                {
+                    Bitmap bitIcon = Icon.ExtractAssociatedIcon(Environment.GetFolderPath(Environment.SpecialFolder.Windows) + "\\explorer.exe").ToBitmap();
+                    using (var memory = new MemoryStream())
+                    {
+                        bitIcon.Save(memory, System.Drawing.Imaging.ImageFormat.Png);
+                        memory.Position = 0;
+                        icon.BeginInit();
+                        icon.StreamSource = memory;
+                        icon.CacheOption = BitmapCacheOption.OnLoad;
+                        icon.EndInit();
+                        icon.Freeze();
+                    }
+                }
+                else // Other app without icon
+                {
+                    icon.BeginInit();
+                    icon.UriSource = new Uri("pack://application:,,,/ClipboardZanager;component/Assets/NoIcon.png", UriKind.RelativeOrAbsolute);
+                    icon.CacheOption = BitmapCacheOption.OnLoad;
+                    icon.EndInit();
+                    icon.Freeze();
+                }
+
+            }
+
             if (string.IsNullOrEmpty(applicationIdentifier))
             {
```

Comment:
If I understand well the code (I didn't tried it yet), I think that this condition is deprecated.